### PR TITLE
RA-1592 : Boostrap as a foundation for Reference Application UI Project.

### PR DESCRIPTION
**Pull request by Akayeshmantha**
**This PR contains changes for the GSoC Project Bootstrap as a foundation for Ref App UI project.
It contains the following changes up to now**

Home Page css changes with media queries.
Login.gsp page bootstrap changes.
Home.gsp page bootstrap changes.
UserApp.gsp page bootstrap changes.
Bootstrap integration(Currently works with internet only).

**This PR will have more commit's when the project is going forward.**

Ticket ID :- RA-1952

### CODE REVIEW

File: omod/src/main/webapp/pages/login.gsp [hunk 1]
```diff
@@ -9,6 +9,9 @@
     <title>${ ui.message("referenceapplication.login.title") }</title>
     <link rel="shortcut icon" type="image/ico" href="/${ ui.contextPath() }/images/openmrs-favicon.ico"/>
     <link rel="icon" type="image/png\" href="/${ ui.contextPath() }/images/openmrs-favicon.png"/>
+    <!-- Latest compiled and minified CSS -->
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.3.1/css/bootstrap.min.css">
```
Comment (dkayiwa) [inline omod/src/main/webapp/pages/login.gsp]: Will this work without an internet connection?

File: omod/src/main/webapp/pages/login.gsp [hunk 2]
```diff
@@ -104,85 +105,109 @@ ${ ui.includeFragment("referenceapplication", "infoAndErrorMessages") }
...
+
+                    <button class="confirm">${ ui.message("referenceapplication.okay") }</button>
+                </div>
+            </div>
+        </div>
     </div>
-</div>
 
+    <!-- jQuery library -->
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.4.0/jquery.min.js"></script>
```
Comment (dkayiwa) [inline omod/src/main/webapp/pages/login.gsp]: Will these work without an internet connection?

File: omod/src/main/compass/sass/home.scss [hunk 3]
```diff
@@ -16,4 +16,56 @@ hr.separator {
...
+}
+
+.homeList a  {
+  font-size: unset;
+  white-space: normal;
+  width: 166px !important;
+  color: #363463;
+}
+
+.homeList a:hover{
```
Comment (suthagar23) [inline omod/src/main/compass/sass/home.scss]: these can go under nested styles in scss.

File: omod/src/main/webapp/pages/home.gsp [hunk 4]
```diff
@@ -1,34 +1,39 @@
 <%
     ui.decorateWith("appui", "standardEmrPage", [ title: ui.message("referenceapplication.home.title") ])
+    ui.includeCss("referenceapplication", "home.css")
```
Comment (suthagar23) [inline omod/src/main/webapp/pages/home.gsp]: is it `home.css` or `home.scss`?

Comment (Akayeshmantha) [reply omod/src/main/webapp/pages/home.gsp]: it is css normally referring after the build.

File: omod/src/main/compass/sass/home.scss (hunk 3 above)
Comment (Akayeshmantha) [reply omod/src/main/compass/sass/home.scss]: done

File: omod/src/main/webapp/pages/login.gsp (hunk 2 above)
Comment (Akayeshmantha) [reply omod/src/main/webapp/pages/login.gsp]: now refereed locally from appui module.

File: omod/src/main/webapp/pages/login.gsp (hunk 1 above)
Comment (Akayeshmantha) [reply omod/src/main/webapp/pages/login.gsp]: locally referenced

**Pull request merged**